test(linter/plugins): add conformance tests

overlookmotel opened the pull request:
Add conformance tester for JS plugins.

The tester runs all of ESLint's test cases for all its built-in rules, but using Oxlint's `RuleTester` in place of ESLint's. i.e. the ESLint rules are run as Oxlint JS plugins.

It outputs the results in `snapshot.md` file.

Majority of the failing test cases are false positives. Later PRs will improve the conformance tester and `RuleTester` so it correctly handles `sourceType`, `jsx`, and TypeScript sources.

These tests do not run in CI. For now it's just useful for me to commit this, while I work on improving the tester.

I'm not sure if inside `apps/oxlint` is the right place for this ultimately, but for now I think it's passable.

overlookmotel commented:
* **#16650** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16650?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16650?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=16650).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented on apps/oxlint/conformance/README.md:
personally i think this should go in tasks to match the other conformance suites - it feels a bit wierd to have a package kind of in a package. I don't feel strongly about this though

overlookmotel replied:
Agreed it may not be the right place. But can we please leave it here while I work on it? It's a bit tricky otherwise.

overlookmotel commented on apps/oxlint/conformance/src/capture.ts:
```diff
@@ -0,0 +1,87 @@
+    testResult.error = err as Error;
+  }
+
+  currentRule!.tests.push(testResult);
```
You and your excessive error-handling... no!

overlookmotel committed:
test(linter/plugins): add conformance tests (#16650)

Add conformance tester for JS plugins.

The tester runs all of ESLint's test cases for all its built-in rules, but using Oxlint's `RuleTester` in place of ESLint's. i.e. the ESLint rules are run as Oxlint JS plugins.

It outputs the results in `snapshot.md` file.

Majority of the failing test cases are false positives. Later PRs will improve the conformance tester and `RuleTester` so it correctly handles `sourceType`, `jsx`, and TypeScript sources.

These tests do not run in CI. For now it's just useful for me to commit this, while I work on improving the tester.

I'm not sure if inside `apps/oxlint` is the right place for this ultimately, but for now I think it's passable.
